src/components/application_manager/doc/SDL.SDD.StateController.dox

```diff
@@ -0,0 +1,73 @@
+/**
+\page application_manager State Controller Detailed Design
+
+## State Controller Design
+- \subpage state_controller_high_level_logic
+
+*/
+
+/**
+\page state_controller_high_level_logic 3 State Controller High Level Logic
+
+## Introduction
+The state controller is responsible for keeping applications in appropriate HMI state
+HMI state contains: 
+ - HMILevel 
+ - AudioStreamingState
+ - VideoStreamingState
+ - SystemContext 
+ 
+SDL should take care of conflicts between the HMI states of different apps. 
+For example, 2 apps cannot be in FULL at the same time. So if one application goes FULL, another one should be moved to BACKGROUND or LIMITED.
+Rules are described in the scope of the proposal https://github.com/smartdevicelink/sdl_evolution/blob/master/proposals/0150-video-streaming-state.md 
+
+SDL should resolve application HMI states to avoid conflicts. There are several triggers for HMI state change:
+ - application activation
+ - application deactivation
+ - Start System Event from HMI (Phone call, TTS, etc ...)
+ - End System Event from HMI 
+ - Resumption
+
+\link application_manager::StateController 
+\endlink is responsible for this.
+
+Event from HMI should change HMI state for registered applications.
+Changing rules described in  https://github.com/smartdevicelink/sdl_evolution/blob/proposals/0150-video-streaming-state.md 
```

Comment:
Seems there are two similar sentences with the same link.

Reply:
removed one